### PR TITLE
model.py: design and create the basic model info class

### LINKED ISSUE #29: design and create the model info base class
We need a model info class to store the model meta and some model metrics.

### PR DESCRIPTION
**Pull request by leyao-daily**
This PR is to fix #29 and it introduces three new classes: ModelInfo, ModelDetails, and ModelMetrics. These classes are designed to encapsulate all relevant information about an Inference model within a inference pipeline.

ModelMetrics holds the performance metrics of a inference learning model, such as accuracy, precision, recall, F1 score, and loss value. Each of these metrics can be accessed using their respective property methods.

ModelDetails encapsulates various details about the model, including the model's name, version, the framework used for training, the target, and the data type of the model.

ModelInfo ties all this information together, containing an instance of ModelDetails, the model's upload date, and an instance of ModelMetrics. Additionally, it includes a unique identifier for each model using Python's built-in uuid library.

### CODE REVIEW

File: cnap/core/model.py [hunk 1]
```diff
@@ -0,0 +1,209 @@
...
+
+    @property
+    def id(self) -> str:
+        """
+        Get the ID of the model (string of UUID).
+        """
+        return self._id
+
+    @id.setter
+    def id(self, new_str) -> None:
```
Comment (Hulongyin) [inline cnap/core/model.py]: Add hint for the `new_str` parameter.

Comment (leyao-daily) [reply cnap/core/model.py]: Fixed.

**Review by Hulongyin — APPROVED** [review]
LGTM. Thanks

**Comment by leyao-daily** [comment]
Closes #29

**Pull request merged**